app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentMPU6050.java

```diff
@@ -0,0 +1,118 @@
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import org.fossasia.pslab.R;
+import org.fossasia.pslab.communication.ScienceLab;
+import org.fossasia.pslab.communication.peripherals.I2C;
+import org.fossasia.pslab.communication.sensors.MPU6050;
+import org.fossasia.pslab.others.ScienceLabCommon;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by asitava on 10/7/17.
+ */
+
+public class SensorFragmentMPU6050 extends Fragment {
+    private ScienceLab scienceLab;
+    private I2C i2c;
+
+    public static SensorFragmentMPU6050 newInstance() {
+        SensorFragmentMPU6050 sensorFragmentMPU6050 = new SensorFragmentMPU6050();
+        return sensorFragmentMPU6050;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        scienceLab = ScienceLabCommon.scienceLab;
+        i2c = scienceLab.i2c;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.sensor_mpu6050, container, false);
+
+        final TextView tvSensorMPU6050ax = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_ax);
+        final TextView tvSensorMPU6050ay = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_ay);
+        final TextView tvSensorMPU6050az = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_az);
+        final TextView tvSensorMPU6050gx = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_gx);
+        final TextView tvSensorMPU6050gy = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_gy);
+        final TextView tvSensorMPU6050gz = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_gz);
+        final TextView tvSensorMPU6050temp = (TextView) view.findViewById(R.id.tv_sensor_mpu6050_temp);
+
+        Spinner spinnerSensorMPU60501 = (Spinner) view.findViewById(R.id.spinner_sensor_mpu6050_1);
+        Spinner spinnerSensorMPU60502 = (Spinner) view.findViewById(R.id.spinner_sensor_mpu6050_2);
+        Spinner spinnerSensorMPU60503 = (Spinner) view.findViewById(R.id.spinner_sensor_mpu6050_3);
+        Spinner spinnerSensorMPU60504 = (Spinner) view.findViewById(R.id.spinner_sensor_mpu6050_4);
+
+        MPU6050 MPU6050 = null;
+        try {
+            MPU6050 = new MPU6050(i2c);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        new Timer().schedule(new TimerTask() {
+            @Override
+            public void run() {
+                getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        MPU6050 MPU6050 = null;
+                        try {
+                            MPU6050 = new MPU6050(i2c);
```

Comment:
Created an MPU6050 object here also and above too. Timer would execute this TimerTask at specified delay again and again. So object is allocated again and again periodically.
And I don't see any code to handle the Timer when I exit back. Doesn't it give any error when you exit before cancelling the Timer?
And you are reading values on UI thread. Not recommended, would result in lagging UI during chart plot. Currently readback value are less, so its not that visible.
Have a look at AsyncTask to implement this or if you can handle data across threads that okay too.